esmvaltool/diag_scripts/seaice_area_extents/seaice_mapextents.py

```diff
@@ -0,0 +1,156 @@
+"""Diagnostic script to plot extent and differences.
+
+based on code from Anton Steketee's COSIMA recipes notebook
+https://cosima-recipes.readthedocs.io/en/latest/Examples/Sea_Ice_Area_Concentration_Volume_with_Obs.html
+"""
+
+import logging
+import os
+import calendar
+from cartopy.crs import SouthPolarStereo
+import xarray as xr
+import xesmf
+import matplotlib.pyplot as plt
+import matplotlib.lines as mlines
+import numpy as np
+import pandas as pd
+
+from esmvaltool.diag_scripts.shared import (run_diagnostic, save_figure,
+                                            save_data)
+from esmvaltool.diag_scripts.shared._base import get_plot_filename
+
+# This part sends debug statements to stdout
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def model_regrid_diff(mod_si, obs_si, cdr, mon, latmax):
+    """Regrid and compute difference grid."""
+    lat_min = mod_si.lat.min().values.item()
+    mod_si = mod_si.where(mod_si.lat < latmax, drop=True)
+    regrd_out = obs_si.where(obs_si.lat > lat_min, drop=True)
+
+    # regrid model data to observations
+    regridder_access_sh = xesmf.Regridder(mod_si.isel(time=0).drop(['i', 'j']),
+                                          regrd_out.isel(time=0),
+                                          'bilinear',
+                                          periodic=True,
+                                          unmapped_to_nan=True)
+
+    model_mean = mod_si.siconc.sel(
+        time=mod_si.siconc.time.dt.month.isin(mon)).mean('time')
+    mod_regrid = regridder_access_sh(model_mean)
+    diff_ds = mod_regrid - cdr
+
+    return diff_ds, mod_regrid
+
+
+def map_diff(mod_si_ls, obs_si, months, cfg, prov):
+    """Create figure mapping extents for models and months."""
+    # get lat max for regridding
+    latmax = obs_si.lat.max().values.item()
+
+    proj = SouthPolarStereo(true_scale_latitude=-70)
+
+    # fig set up, width for 2 models, check len mod_si_ls
+    figure = plt.figure(figsize=(9, len(months) * 4))
+
+    for j, mon in enumerate(months):
+        cdr = obs_si.siconc.sel(
```

Comment:
I noticed that the variable names in these scripts are rather terse, e.g. what does `cdr` mean? How about using longer, more descriptive names?